sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultService.java

```diff
@@ -70,116 +53,82 @@ public DefaultService(File pathToYamlSpecification) throws Exception {
     }
 
     public DefaultService(RawServiceSpecification rawServiceSpecification) throws Exception {
-        CapabilityValidator capabilityValidator = new CapabilityValidator(new Capabilities(new DcosCluster()));
-        this.serviceSpec = YAMLServiceSpecFactory.generateServiceSpec(rawServiceSpecification);
-        capabilityValidator.validate(this.serviceSpec);
-
-        init();
-        this.plans = generatePlansFromRawSpec(rawServiceSpecification);
-        register(serviceSpec, this.plans);
-        unlock();
+        this(DefaultScheduler.newBuilder(YAMLServiceSpecFactory.generateServiceSpec(rawServiceSpecification))
+                .setPlansFrom(rawServiceSpecification));
     }
 
-    public DefaultService(ServiceSpec serviceSpecification) {
-        this(serviceSpecification, Collections.emptyList());
+    public DefaultService(ServiceSpec serviceSpecification, Collection<Plan> plans) throws Exception {
+        this(DefaultScheduler.newBuilder(serviceSpecification).setPlans(plans));
     }
 
-    public DefaultService(ServiceSpec serviceSpec, Collection<Plan> plans) {
-        this.serviceSpec = serviceSpec;
-        init();
-        this.plans = plans;
-        register(serviceSpec, this.plans);
+    public DefaultService(DefaultScheduler.Builder schedulerBuilder) throws Exception {
+        this.schedulerBuilder = schedulerBuilder;
+
+        CuratorFramework curatorClient = CuratorFrameworkFactory.newClient(
+                schedulerBuilder.getServiceSpec().getZookeeperConnection(), CuratorUtils.getDefaultRetry());
+        curatorClient.start();
```

Comment:
Ok for the reference to `curatorClient` to become unreachable after we leave this function scope, since we'll never be able to close it? I guess there's no issue with sockets being held open here, since a crash will take care of that anyway.